knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -265,11 +280,13 @@ func (r *ReconcileKnativeKafka) deleteResources(manifest *mf.Manifest, instance
 		return nil
 	}
 	log.Info("Deleting resources in manifest")
-	if err := manifest.Delete(); err != nil {
-		// TODO: any conditions?
-		return fmt.Errorf("failed to apply manifest: %w", err)
+	if err := manifest.Filter(mf.NoCRDs, not(mf.Any(role, rolebinding))).Delete(); err != nil {
+		return fmt.Errorf("failed to remove non-crd/non-rbac resources: %w", err)
+	}
+	// Delete Roles last, as they may be useful for human operators to clean up.
+	if err := manifest.Filter(mf.Any(role, rolebinding)).Delete(); err != nil {
```

Comment:
nit: `mf.Any(role, rolebinding)` is used three times here. Maybe create a var above with `rbac` that has that too?

Reply:
+1. Done